# ğŸ¨ Samanet Professional Dashboard Design System - Cursor Rules

## âœ… **ENTERPRISE-GRADE DASHBOARD SYSTEM ACTIVE**

**Date Updated:** January 2025  
**Status:** ğŸ”¥ **ACTIVE & AUTO-APPLYING PROFESSIONAL STANDARDS**

---

## ğŸ“ **Updated Files Overview**

### **1. `MANDATORY-DESIGN.mdc`** â­ **MAIN PROFESSIONAL SYSTEM**
- **Status:** âœ… **ENTERPRISE-GRADE DASHBOARD RULES**
- **Priority:** 1000
- **Contains:** Complete professional dashboard system with enterprise-grade components
- **Based On:** Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard.css Ùˆ users.css

### **2. `samanat-design-standards-fa.mdc`** ğŸ‡®ğŸ‡· **PERSIAN PROFESSIONAL REFERENCE**
- **Status:** âœ… **ACTIVE PROFESSIONAL GUIDELINES**
- **Priority:** 200
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **3. `SEARCH-FILTER-SYSTEM.mdc`** ğŸ” **ADVANCED SEARCH SYSTEM** âœ¨
- **Status:** âœ… **MANDATORY - APPLY TO ALL SEARCH FEATURES**
- **Priority:** 300
- **Purpose:** Professional real-time multi-word search system
- **Contains:** Complete search implementation with highlighting, debouncing, mobile optimization

### **4. `PROJECT-DOCUMENTATION-MANAGER.mdc`** ğŸ“š **AUTO-DOCUMENTATION SYSTEM** âœ¨
- **Status:** âœ… **MANDATORY - AUTO-MAINTAIN ALL DOCUMENTATION**
- **Priority:** 400
- **Purpose:** Automatic documentation maintenance and standards
- **Contains:** Complete documentation management, auto-update triggers, quality assurance checklist

### **5. `DEBUGGING-RULES.mdc`** ğŸ› **DEBUGGING & TROUBLESHOOTING**
- **Status:** âœ… **ACTIVE DEBUGGING GUIDELINES**
- **Priority:** 150
- **Purpose:** Common issues and solutions
- **Contains:** Theme management fixes, function redeclaration solutions

### **6. `samanat-coding-standard.mdc`** ğŸ“ **CODING STANDARDS**
- **Status:** âœ… **ACTIVE CODING GUIDELINES**
- **Priority:** 500
- **Purpose:** PHP, JavaScript, and general coding standards
- **Contains:** Complete development guidelines with theme management rules

### **7. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Status:** âœ… **ACTIVE DESIGN GUIDELINES**
- **Priority:** 300
- **Purpose:** Complete design system documentation
- **Contains:** Glassmorphism, animations, modern components

### **8. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Language:** Persian/Farsi
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **9. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **10. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **11. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **12. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **13. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **14. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **15. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **16. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **17. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **18. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **19. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

### **20. `samanat-design-system.mdc`** ğŸ¨ **DESIGN SYSTEM REFERENCE**
- **Purpose:** Persian professional dashboard standards
- **Contains:** Complete Persian enterprise-grade guidelines

---

## ğŸ¯ **Professional Dashboard Standards That NOW WORK**

### **MANDATORY Professional Components:**
```css
/* MANDATORY CSS Variables - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard.css */
:root {
    /* Professional Color Palette - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
    --primary: #5E3AEE;
    --primary-light: #7C5CFF;
    --primary-dark: #4527A0;
    
    --gray-50: #FAFAFA;
    --gray-100: #F4F4F5;
    --gray-200: #E4E4E7;
    --gray-300: #D4D4D8;
    --gray-400: #A1A1AA;
    --gray-500: #71717A;
    --gray-600: #52525B;
    --gray-700: #3F3F46;
    --gray-800: #27272A;
    --gray-900: #18181B;
    
    --success: #10B981;
    --warning: #F59E0B;
    --danger: #EF4444;
    --info: #3B82F6;
    
    /* Spacing System - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
    --space-0: 0;
    --space-1: 0.25rem;
    --space-2: 0.5rem;
    --space-3: 0.75rem;
    --space-4: 1rem;
    --space-5: 1.25rem;
    --space-6: 1.5rem;
    --space-8: 2rem;
    
    /* Typography - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
    --font-sans: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Vazirmatn', sans-serif;
    --font-mono: 'SF Mono', Monaco, monospace;
    
    /* Shadows - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
    --shadow-xs: 0 1px 2px 0 rgb(0 0 0 / 0.05);
    --shadow-sm: 0 1px 3px 0 rgb(0 0 0 / 0.1);
    --shadow-md: 0 4px 6px -1px rgb(0 0 0 / 0.1);
    --shadow-lg: 0 10px 15px -3px rgb(0 0 0 / 0.1);
    
    /* Border Radius - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
    --radius-sm: 0.375rem;
    --radius-md: 0.5rem;
    --radius-lg: 0.75rem;
    --radius-xl: 1rem;
    
    /* Transitions - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
    --transition: all 150ms ease;
}

/* Dark Mode Variables - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
[data-theme="dark"] {
    --gray-50: #18181B;
    --gray-100: #27272A;
    --gray-200: #3F3F46;
    --gray-300: #52525B;
    --gray-400: #71717A;
    --gray-500: #A1A1AA;
    --gray-600: #D4D4D8;
    --gray-700: #E4E4E7;
    --gray-800: #F4F4F5;
    --gray-900: #FAFAFA;
}

/* Professional Dashboard Wrapper */
.dashboard-pro {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
}

/* Professional Header (60px) */
.dashboard-header {
    height: 60px;
    background: white;
    border-bottom: 1px solid var(--gray-200);
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 var(--space-6);
    position: sticky;
    top: 0;
    z-index: 40;
}

/* Theme Toggle Button - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
.theme-toggle,
.btn-icon {
    width: 40px;
    height: 40px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    background: var(--gray-100);
    border: 1px solid var(--gray-200);
    color: var(--gray-600);
    cursor: pointer;
    transition: var(--transition);
}

/* User Profile - Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard */
.user-profile {
    width: 40px;
    height: 40px;
    border-radius: 50%;
    background: var(--primary);
    color: white;
    display: flex;
    align-items: center;
    justify-content: center;
    font-weight: 600;
    font-size: 14px;
}

/* Professional Stats Cards */
.stat-card-pro {
    background: white;
    border: 1px solid var(--gray-200);
    border-radius: var(--radius-lg);
    padding: var(--space-4);
    transition: var(--transition);
}

/* Professional Tables */
.table-container {
    background: white;
    border: 1px solid var(--gray-200);
    border-radius: var(--radius-lg);
    overflow: hidden;
}

/* Professional Side Panels */
.panel {
    background: white;
    border: 1px solid var(--gray-200);
    border-radius: var(--radius-lg);
    overflow: hidden;
}
```

### **MANDATORY Professional HTML Structure:**
```html
<div class="dashboard-pro">
    <header class="dashboard-header">
        <div class="header-content">
            <h1 class="header-title">Ø¹Ù†ÙˆØ§Ù† ØµÙØ­Ù‡</h1>
        </div>
        <div class="header-actions">
            <button class="theme-toggle" onclick="toggleTheme()">
                <i class="fas fa-moon"></i>
            </button>
            <div class="user-profile">Ú©</div>
        </div>
    </header>

    <div class="dashboard-content">
        <div class="stats-row">
            <div class="stat-card-pro">
                <!-- Professional stats -->
            </div>
        </div>

        <div class="dashboard-grid">
            <div class="main-column">
                <div class="table-container">
                    <!-- Professional table -->
                </div>
            </div>
            <div class="side-column">
                <div class="panel">
                    <!-- Professional panels -->
                </div>
            </div>
        </div>
    </div>
</div>
```

### **MANDATORY Perfect Dark Mode:**
```css
/* Perfect Dark Mode Support */
[data-theme="dark"] .dashboard-header {
    background: var(--gray-100);
    border-color: var(--gray-200);
}

[data-theme="dark"] .stat-card-pro,
[data-theme="dark"] .table-container,
[data-theme="dark"] .panel {
    background: var(--gray-100);
    border-color: var(--gray-200);
}

[data-theme="dark"] .task-text {
    color: var(--gray-800) !important;
}

[data-theme="dark"] .panel-title {
    color: var(--gray-800);
}
```

---

## ğŸ§ª **Test the Professional Auto-Apply**

To verify professional rules are working:

1. **Create a new dashboard/page file**
2. **Cursor should automatically suggest:**
   - `.dashboard-pro` wrapper
   - `.dashboard-header` with 60px height
   - `.stat-card-pro` for statistics
   - `.table-container` for tables
   - `.panel` for side panels
   - Perfect dark mode with `var(--gray-800)` text

3. **Expected result:** Enterprise-grade dashboard matching dashboard.css and users.css exactly

---

## ğŸ“Š **Professional Quality Standards**

### **Enterprise-Grade Requirements:**
1. **Design Quality**: Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard.css Ùˆ users.css
2. **Data Density**: Maximum info, minimum space  
3. **Typography**: Professional hierarchy Ø¨Ø§ ÙÙˆÙ†Øª `-apple-system, BlinkMacSystemFont, 'Segoe UI', 'Vazirmatn'`
4. **Performance**: Lightweight and fast
5. **Accessibility**: WCAG 2.1 compliant
6. **Mobile**: Perfect responsive design
7. **Themes**: Flawless light/dark mode

### **Mandatory Features:**
- **Header**: 60px sticky, minimal design
- **Stats**: 4-column compact grid
- **Tables**: Professional with mobile cards
- **Panels**: Tasks, activities, urgent items
- **Theme**: 40px toggle button (Ø¯Ø§ÛŒØ±Ù‡â€ŒØ§ÛŒ)
- **Profile**: 40px user circle
- **Animations**: Smooth micro-interactions

---

## âœ… **Success Indicators**

When professional auto-apply works, you'll see:
- âœ… Automatic suggestion of `.dashboard-pro` wrapper
- âœ… Professional header with 60px height
- âœ… Enterprise-grade stats cards
- âœ… Professional table styling
- âœ… Perfect dark mode readability
- âœ… Mobile-responsive design
- âœ… Smooth micro-interactions

---

## ğŸ¯ **How Professional Dashboard System Works**

When you create or modify ANY dashboard, Cursor AI will **automatically**:

1. âœ… Apply enterprise-grade dashboard design
2. âœ… Use professional color system  
3. âœ… Implement data-dense layouts
4. âœ… Add smooth micro-interactions
5. âœ… Ensure perfect mobile responsiveness
6. âœ… Include flawless dark mode
7. âœ… Follow accessibility standards

## ğŸš€ **What You Get Automatically**

### **Every Dashboard Will Have:**
- ğŸ¨ **Enterprise-Grade Design** (Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard.css Ùˆ users.css)
- âœ¨ **Data-Dense Layouts** with maximum information density
- ğŸ“± **Perfect Mobile Responsiveness** from 320px to 4K
- ğŸ¯ **Professional Typography** with clear hierarchy
- âš¡ **Optimized Performance** with lightweight CSS
- â™¿ **Accessibility Compliance** (WCAG 2.1)
- ğŸŒ™ **Flawless Dark Mode** with perfect contrast

### **Required Professional Classes:**
- `.dashboard-pro` - Main dashboard wrapper
- `.dashboard-header` - Professional 60px header
- `.stat-card-pro` - Enterprise-grade stats
- `.table-container` - Professional tables
- `.panel` - Side panels for tasks/activities
- `.theme-toggle` - 40px icon buttons (Ø¯Ø§ÛŒØ±Ù‡â€ŒØ§ÛŒ)
- `.user-profile` - 40px user circle

### **Required Professional Structure:**
```html
<div class="dashboard-pro">
    <!-- Professional 60px header -->
    <!-- Compact 4-column stats grid -->
    <!-- Professional table with mobile cards -->
    <!-- Side panels for tasks and activities -->
</div>
```

### **Required JavaScript:**
```javascript
// Professional progressive animations
// Table hover effects
// Theme persistence testing
```

## âœ… **Professional Quality Guaranteed**

Every dashboard will automatically meet these enterprise standards:
- âœ… Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard.css Ùˆ users.css design quality
- âœ… Data-dense efficient layouts
- âœ… Mobile-first responsive design
- âœ… Fast loading performance
- âœ… Perfect dark mode contrast
- âœ… Accessibility compliance

## ğŸ¨ **Professional Design Philosophy Applied**

1. **Enterprise-Grade Design** - Commercial quality standards
2. **Data Density** - Maximum information, minimum space
3. **Professional Typography** - Clear hierarchy and readability
4. **Micro-Interactions** - Subtle, delightful animations
5. **Mobile-First** - Perfect responsive from 320px to 4K
6. **Performance First** - Lightweight and optimized
7. **Accessibility First** - WCAG 2.1 compliance

## ğŸ”§ **How to Customize Professional System**

If you need to modify the professional dashboard system:

1. Edit `MANDATORY-DESIGN.mdc` for new professional components
2. Update `design-prompt.mdc` for new professional enforcement rules
3. Modify `samanat-design-standards-fa.mdc` for Persian professional updates

Changes will be automatically applied to all new dashboards.

## ğŸ’¡ **Important Note About Professional Standards**

All rule files enforce enterprise-grade professional standards ensuring every dashboard matches the quality of dashboard.css and users.css exactly. The system automatically applies:

- **Professional Color System**: Enterprise-grade grays and accent colors
- **Data Density**: Maximum information in minimum space
- **Typography**: Professional hierarchy with system fonts
- **Performance**: Lightweight CSS with optimized animations
- **Accessibility**: WCAG 2.1 compliant with perfect contrast ratios

## ğŸ“ **Support**

The professional dashboard system is self-enforcing through Cursor AI. Every dashboard you create will automatically follow these enterprise-grade professional standards without any manual intervention.

---

**ğŸ¨ Status:** âœ… ACTIVE - Professional Auto-Applied to ALL Dashboards  
**ğŸ“… Version:** 5.0 Enterprise Grade  
**ğŸ¯ Scope:** Universal application to entire project with enterprise-grade quality matching dashboard.css and users.css exactly 

# ğŸ“š Ø±Ø§Ù‡Ù†Ù…Ø§ÛŒ Ø³ÛŒØ³ØªÙ… Ù‚ÙˆØ§Ù†ÛŒÙ† Ø·Ø±Ø§Ø­ÛŒ Ø³Ø§Ù…Ø§Ù†Øª
## Ù…Ø±Ú©Ø² Ù…Ø¯ÛŒØ±ÛŒØª Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯Ù‡Ø§ÛŒ ØªÙˆØ³Ø¹Ù‡ Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ

**Ù†Ø³Ø®Ù‡:** 6.0  
**ØªØ§Ø±ÛŒØ® Ø¨Ø±ÙˆØ²Ø±Ø³Ø§Ù†ÛŒ:** 1404/10/15  
**ÙˆØ¶Ø¹ÛŒØª:** âœ… ÙØ¹Ø§Ù„ Ùˆ Ú©Ø§Ù…Ù„  

---

## ğŸ“‹ **ÙØ§ÛŒÙ„â€ŒÙ‡Ø§ÛŒ Ù‚ÙˆØ§Ù†ÛŒÙ† Ù…ÙˆØ¬ÙˆØ¯**

### **1. MANDATORY-DESIGN.mdc** â­ (Ø§ÙˆÙ„ÙˆÛŒØª 1000)
**Ù‡Ø¯Ù:** Ù‚ÙˆØ§Ù†ÛŒÙ† Ø§Ø¬Ø¨Ø§Ø±ÛŒ Ø·Ø±Ø§Ø­ÛŒ Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ  
**Ù…Ø­ØªÙˆØ§:**
- Ù…ØªØºÛŒØ±Ù‡Ø§ÛŒ CSS Ø§Ù„Ø²Ø§Ù…ÛŒ (Ø¯Ù‚ÛŒÙ‚Ø§Ù‹ Ù…Ø·Ø§Ø¨Ù‚ dashboard.css)
- Ú©Ù„Ø§Ø³â€ŒÙ‡Ø§ÛŒ Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ (.dashboard-pro, .stat-card-pro, .data-table)
- **Ù‚ÙˆØ§Ù†ÛŒÙ† Theme Management** (Ø¬Ø¯ÛŒØ¯ âœ¨)
- Dark Mode Ú©Ø§Ù…Ù„
- Responsive Design
- HTML Structure Ø§Ù„Ø²Ø§Ù…ÛŒ

**ğŸŒ™ Ø¬Ø¯ÛŒØ¯ Ø¯Ø± Ø§ÛŒÙ† Ù†Ø³Ø®Ù‡:**
- Ù‚ÙˆØ§Ù†ÛŒÙ† Ø§Ø¬Ø¨Ø§Ø±ÛŒ theme management
- Ù…Ù…Ù†ÙˆØ¹ÛŒØª Ø§ÛŒØ¬Ø§Ø¯ `toggleTheme()` Ù…Ø­Ù„ÛŒ
- Ø§Ù„Ø²Ø§Ù… Ø§Ø³ØªÙØ§Ø¯Ù‡ Ø§Ø² localStorage key `samanat_theme`
- Ø±Ø§Ù‡Ù†Ù…Ø§ÛŒ ØµØ­ÛŒØ­ theme button

### **2. samanat-coding-standard.mdc** (Ø§ÙˆÙ„ÙˆÛŒØª 500)
**Ù‡Ø¯Ù:** Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯Ù‡Ø§ÛŒ Ú©Ø¯Ù†ÙˆÛŒØ³ÛŒ PHP Ùˆ JavaScript  
**Ù…Ø­ØªÙˆØ§:**
- Ù‚ÙˆØ§Ù†ÛŒÙ† PHP (PSR-12, Clean Architecture)
- Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯Ù‡Ø§ÛŒ JavaScript ES6+
- **Ù‚ÙˆØ§Ù†ÛŒÙ† Theme Management** (Ø¬Ø¯ÛŒØ¯ âœ¨)
- Ù…Ø¯ÛŒØ±ÛŒØª Ø®Ø·Ø§ Ùˆ Exception
- Ø§Ù…Ù†ÛŒØª Ùˆ Ø§Ø¹ØªØ¨Ø§Ø±Ø³Ù†Ø¬ÛŒ

**ğŸŒ™ Ø¬Ø¯ÛŒØ¯ Ø¯Ø± Ø§ÛŒÙ† Ù†Ø³Ø®Ù‡:**
- Ù‚ÙˆØ§Ù†ÛŒÙ† localStorage ØµØ­ÛŒØ­
- Ù†Ù…ÙˆÙ†Ù‡â€ŒÙ‡Ø§ÛŒ Ú©Ø¯ Ø§Ø´ØªØ¨Ø§Ù‡ Ùˆ ØµØ­ÛŒØ­
- Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯Ù‡Ø§ÛŒ Ù…Ø¯ÛŒØ±ÛŒØª ØªÙ…

### **3. DEBUGGING-RULES.mdc** (Ø¬Ø¯ÛŒØ¯ âœ¨)
**Ù‡Ø¯Ù:** Ø±Ø§Ù‡Ù†Ù…Ø§ÛŒ Ø®Ø·Ø§ÛŒØ§Ø¨ÛŒ Ùˆ Ø±ÙØ¹ Ù…Ø´Ú©Ù„Ø§Øª  
**Ù…Ø­ØªÙˆØ§:**
- Ù…Ø´Ú©Ù„Ø§Øª Ø±Ø§ÛŒØ¬ Theme Management
- Ø®Ø·Ø§ÛŒ Function Redeclaration
- Ù…Ø´Ú©Ù„Ø§Øª CSS Variables
- Ø±Ø§Ù‡â€ŒØ­Ù„â€ŒÙ‡Ø§ÛŒ Database Model
- Ú†Ú©â€ŒÙ„ÛŒØ³Øª Ù¾ÛŒØ´ Ø§Ø² Deploy

### **4. ultra-modern-design-system.mdc** (Ø§ÙˆÙ„ÙˆÛŒØª 100)
**Ù‡Ø¯Ù:** Ø³ÛŒØ³ØªÙ… Ø·Ø±Ø§Ø­ÛŒ Ù…Ø¯Ø±Ù† Ú©Ø§Ù…Ù„  
**Ù…Ø­ØªÙˆØ§:**
- Glass Morphism
- Ø§Ù†ÛŒÙ…ÛŒØ´Ù†â€ŒÙ‡Ø§ÛŒ Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ
- Ú©Ø§Ù…Ù¾ÙˆÙ†Ù†Øªâ€ŒÙ‡Ø§ÛŒ Ù…Ø¯Ø±Ù†
- Auto-Apply Rules

### **5. design-prompt.mdc** (Ø§ÙˆÙ„ÙˆÛŒØª 200)
**Ù‡Ø¯Ù:** Ø±Ø§Ù‡Ù†Ù…Ø§ÛŒ Ø§Ø¹Ù…Ø§Ù„ Ø®ÙˆØ¯Ú©Ø§Ø± Ø·Ø±Ø§Ø­ÛŒ  
**Ù…Ø­ØªÙˆØ§:**
- Ù‚ÙˆØ§Ù†ÛŒÙ† Enterprise-Grade
- Ú†Ú©â€ŒÙ„ÛŒØ³Øª Ø§Ù„Ø²Ø§Ù…ÛŒ
- Ù†Ù…ÙˆÙ†Ù‡ Ú©Ø¯Ù‡Ø§ÛŒ Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ

### **6. samanat-design-system.mdc** (Ø§ÙˆÙ„ÙˆÛŒØª 300)
**Ù‡Ø¯Ù:** Ø³ÛŒØ³ØªÙ… Ø·Ø±Ø§Ø­ÛŒ Ø§Ø®ØªØµØ§ØµÛŒ Ø³Ø§Ù…Ø§Ù†Øª  
**Ù…Ø­ØªÙˆØ§:**
- Ú©Ø§Ù…Ù¾ÙˆÙ†Ù†Øªâ€ŒÙ‡Ø§ÛŒ Ø§Ø®ØªØµØ§ØµÛŒ
- Ø§Ù„Ú¯ÙˆÙ‡Ø§ÛŒ Ø·Ø±Ø§Ø­ÛŒ
- Ø±Ù†Ú¯â€ŒØ¨Ù†Ø¯ÛŒ Ùˆ Typography

### **7. SEARCH-FILTER-SYSTEM.mdc** ğŸ” **ADVANCED SEARCH SYSTEM** âœ¨
- **Status:** âœ… **MANDATORY - APPLY TO ALL SEARCH FEATURES**
- **Priority:** 300
- **Purpose:** Professional real-time multi-word search system
- **Contains:** Complete search implementation with highlighting, debouncing, mobile optimization

### **8. PROJECT-DOCUMENTATION-MANAGER.mdc** ğŸ“š **NEW: AUTO-DOCUMENTATION SYSTEM** âœ¨
- **Status:** âœ… **MANDATORY - AUTO-MAINTAIN ALL DOCUMENTATION**
- **Priority:** 400
- **Purpose:** Automatic documentation maintenance and standards
- **Contains:** Complete documentation management, auto-update triggers, quality assurance checklist

### **9. `DEBUGGING-RULES.mdc`** ğŸ› **DEBUGGING & TROUBLESHOOTING**
- **Status:** âœ… **ACTIVE DEBUGGING GUIDELINES**
- **Priority:** 150
- **Purpose:** Common issues and solutions
- **Contains:** Theme management fixes, function redeclaration solutions

---

## ğŸŒ™ **Ø¨Ø±ÙˆØ²Ø±Ø³Ø§Ù†ÛŒâ€ŒÙ‡Ø§ÛŒ Theme Management**

### **Ù…Ø´Ú©Ù„ Ø­Ù„ Ø´Ø¯Ù‡:**
Ø¯Ø± ØµÙØ­Ù‡ Ø¯Ø±Ø®ÙˆØ§Ø³Øªâ€ŒÙ‡Ø§ÛŒ Ø­ÙˆØ§Ù„Ù‡ØŒ Ø­Ø§Ù„Øª Ø´Ø¨ Ùˆ Ø±ÙˆØ² Ø§Ø² localStorage Ø¨Ø§Ø±Ú¯Ø°Ø§Ø±ÛŒ Ù†Ù…ÛŒâ€ŒØ´Ø¯.

### **Ø¹Ù„Øªâ€ŒÙ‡Ø§ÛŒ Ù…Ø´Ú©Ù„:**
1. **ØªØ¯Ø§Ø®Ù„ JavaScript:** ØªØ§Ø¨Ø¹ `toggleTheme()` Ù…Ø­Ù„ÛŒ Ø¨Ø§ `theme-system.js`
2. **localStorage keys Ù…Ø®ØªÙ„Ø·:** `theme`, `samanat_theme`, `samanet_theme`
3. **Ø¨Ø§Ø±Ú¯Ø°Ø§Ø±ÛŒ Ù…Ø¶Ø§Ø¹Ù:** theme-system.js Ø¯Ùˆ Ø¨Ø§Ø± Ø¨Ø§Ø±Ú¯Ø°Ø§Ø±ÛŒ Ù…ÛŒâ€ŒØ´Ø¯

### **Ø±Ø§Ù‡â€ŒØ­Ù„â€ŒÙ‡Ø§ÛŒ Ø§Ø¹Ù…Ø§Ù„ Ø´Ø¯Ù‡:**
1. **Ø­Ø°Ù JavaScript Ù…Ø­Ù„ÛŒ** Ø§Ø² views
2. **ÛŒÚ©Ù¾Ø§Ø±Ú†Ù‡â€ŒØ³Ø§Ø²ÛŒ localStorage keys** Ø¨Ù‡ `samanat_theme`
3. **Ø¨Ù‡Ø¨ÙˆØ¯ Ø¨Ø§Ø±Ú¯Ø°Ø§Ø±ÛŒ** - theme-system.js Ø¯Ø± `<head>`
4. **Ø¨Ø±ÙˆØ²Ø±Ø³Ø§Ù†ÛŒ Ù‚ÙˆØ§Ù†ÛŒÙ†** Ø¯Ø± ÙØ§ÛŒÙ„â€ŒÙ‡Ø§ÛŒ .mdc

---

## ğŸ¯ **Ù‚ÙˆØ§Ù†ÛŒÙ† Ø¬Ø¯ÛŒØ¯ Theme Management**

### **âŒ NEVER DO:**
```javascript
// âŒ Don't create local toggleTheme functions
function toggleTheme() {
    localStorage.setItem('theme', newTheme); // Wrong key!
}

// âŒ Don't use wrong localStorage keys
localStorage.getItem('theme')
localStorage.getItem('samanet_theme') // Typo!
```

### **âœ… ALWAYS DO:**
```javascript
// âœ… Use central theme system
// Theme system handled by theme-system.js - no local override needed

// âœ… Use correct localStorage key
localStorage.getItem('samanat_theme')

// âœ… Use global toggleTheme function
<button onclick="toggleTheme()">Toggle Theme</button>
```

---

## ğŸ“Š **Ø¢Ù…Ø§Ø± Ø§Ø³ØªÙØ§Ø¯Ù‡ Ø§Ø² Ù‚ÙˆØ§Ù†ÛŒÙ†**

### **âœ… Ù…ÙˆÙÙ‚ÛŒØªâ€ŒÙ‡Ø§ÛŒ Ø§Ø®ÛŒØ±:**
- âœ… **ØµÙØ­Ù‡ Ø¯Ø±Ø®ÙˆØ§Ø³Øªâ€ŒÙ‡Ø§:** Ø¨Ø§Ø²Ø·Ø±Ø§Ø­ÛŒ Ú©Ø§Ù…Ù„ Ù…Ø·Ø§Ø¨Ù‚ dashboard.css
- âœ… **Theme System:** ÛŒÚ©Ù¾Ø§Ø±Ú†Ù‡â€ŒØ³Ø§Ø²ÛŒ Ú©Ø§Ù…Ù„
- âœ… **Ù…Ø´Ú©Ù„ Function Redeclaration:** Ø­Ù„ Ø´Ø¯Ù‡
- âœ… **CSS Variables:** Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯ Ø´Ø¯Ù‡

### **ğŸ“ˆ Ù¾ÛŒØ´Ø±ÙØª Ù¾Ø±ÙˆÚ˜Ù‡:**
- **ÙØ§Ø² 1:** âœ… Dashboard Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ
- **ÙØ§Ø² 2:** âœ… Ù…Ø¯ÛŒØ±ÛŒØª Ú©Ø§Ø±Ø¨Ø±Ø§Ù†
- **ÙØ§Ø² 3:** âœ… Ø¯Ø±Ø®ÙˆØ§Ø³Øªâ€ŒÙ‡Ø§ÛŒ Ø­ÙˆØ§Ù„Ù‡
- **ÙØ§Ø² 4:** ğŸš§ Ø¯Ø± Ø­Ø§Ù„ ØªÙˆØ³Ø¹Ù‡

---

## ğŸ”§ **Ù†Ø­ÙˆÙ‡ Ø§Ø³ØªÙØ§Ø¯Ù‡**

### **Ø¨Ø±Ø§ÛŒ ØªÙˆØ³Ø¹Ù‡â€ŒØ¯Ù‡Ù†Ø¯Ú¯Ø§Ù†:**
1. Ù‡Ù…ÛŒØ´Ù‡ Ø§Ø² `MANDATORY-DESIGN.mdc` Ø´Ø±ÙˆØ¹ Ú©Ù†ÛŒØ¯
2. Ù‚ÙˆØ§Ù†ÛŒÙ† `samanat-coding-standard.mdc` Ø±Ø§ Ø±Ø¹Ø§ÛŒØª Ú©Ù†ÛŒØ¯
3. Ø¯Ø± ØµÙˆØ±Øª Ù…Ø´Ú©Ù„ØŒ `DEBUGGING-RULES.mdc` Ø±Ø§ Ù…Ø·Ø§Ù„Ø¹Ù‡ Ú©Ù†ÛŒØ¯
4. Ø¨Ø±Ø§ÛŒ Ø·Ø±Ø§Ø­ÛŒ Ù…Ø¯Ø±Ù†ØŒ `ultra-modern-design-system.mdc` Ø±Ø§ Ø¨Ø¨ÛŒÙ†ÛŒØ¯

### **Ø¨Ø±Ø§ÛŒ Cursor AI:**
- Ù‡Ù…Ù‡ Ù‚ÙˆØ§Ù†ÛŒÙ† Ø¨ØµÙˆØ±Øª Ø®ÙˆØ¯Ú©Ø§Ø± Ø§Ø¹Ù…Ø§Ù„ Ù…ÛŒâ€ŒØ´ÙˆÙ†Ø¯
- Ø¯Ø± ØµÙˆØ±Øª ØªØ¯Ø§Ø®Ù„ØŒ Ø§ÙˆÙ„ÙˆÛŒØª Ø¨Ø§ `MANDATORY-DESIGN.mdc` Ø§Ø³Øª
- Ù‡Ø± ØµÙØ­Ù‡ Ø¬Ø¯ÛŒØ¯ Ø¨Ø§ÛŒØ¯ Ù…Ø·Ø§Ø¨Ù‚ Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯Ù‡Ø§ Ø¨Ø§Ø´Ø¯

---

## ğŸ“ **Ú†Ú©â€ŒÙ„ÛŒØ³Øª ØªÙˆØ³Ø¹Ù‡**

### **Ù‚Ø¨Ù„ Ø§Ø² Ø´Ø±ÙˆØ¹ Ù‡Ø± ØµÙØ­Ù‡:**
- [ ] Ù…Ø·Ø§Ù„Ø¹Ù‡ `MANDATORY-DESIGN.mdc`
- [ ] Ø¨Ø±Ø±Ø³ÛŒ Ù†Ù…ÙˆÙ†Ù‡â€ŒÙ‡Ø§ÛŒ Ù…ÙˆØ¬ÙˆØ¯
- [ ] Ø¢Ù…Ø§Ø¯Ù‡â€ŒØ³Ø§Ø²ÛŒ CSS variables
- [ ] ØªØ¹Ø±ÛŒÙ Ø³Ø§Ø®ØªØ§Ø± HTML

### **Ø­ÛŒÙ† ØªÙˆØ³Ø¹Ù‡:**
- [ ] Ø±Ø¹Ø§ÛŒØª Ú©Ù„Ø§Ø³â€ŒÙ‡Ø§ÛŒ Ø§Ù„Ø²Ø§Ù…ÛŒ
- [ ] ØªØ³Øª responsive design
- [ ] Ø¨Ø±Ø±Ø³ÛŒ dark mode
- [ ] ØªØ³Øª theme persistence

### **Ù¾Ø³ Ø§Ø² ØªÚ©Ù…ÛŒÙ„:**
- [ ] ØªØ³Øª Ø¯Ø± Ù…Ø±ÙˆØ±Ú¯Ø±Ù‡Ø§ÛŒ Ù…Ø®ØªÙ„Ù
- [ ] Ø¨Ø±Ø±Ø³ÛŒ console errors
- [ ] ØªØ³Øª accessibility
- [ ] Ù…Ø³ØªÙ†Ø¯Ø³Ø§Ø²ÛŒ ØªØºÛŒÛŒØ±Ø§Øª

---

## ğŸš€ **Ø¨Ø±Ù†Ø§Ù…Ù‡ Ø¢ÛŒÙ†Ø¯Ù‡**

### **Ù†Ø³Ø®Ù‡ 7.0 (Ø¢ÛŒÙ†Ø¯Ù‡ Ù†Ø²Ø¯ÛŒÚ©):**
- Ù‚ÙˆØ§Ù†ÛŒÙ† Ù…Ø¯ÛŒØ±ÛŒØª Ø§Ø³Ù†Ø§Ø¯
- Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯Ù‡Ø§ÛŒ API
- Ø±Ø§Ù‡Ù†Ù…Ø§ÛŒ Performance

### **Ù†Ø³Ø®Ù‡ 8.0 (Ø¢ÛŒÙ†Ø¯Ù‡ Ù…ÛŒØ§Ù†Ù‡):**
- Ù‚ÙˆØ§Ù†ÛŒÙ† Ø³ÛŒØ³ØªÙ… ØªØ§ÛŒÛŒØ¯
- Workflow Management
- Advanced Components

---

**ğŸ¯ Ù‡Ø¯Ù Ù†Ù‡Ø§ÛŒÛŒ:** Ø³ÛŒØ³ØªÙ… Ø·Ø±Ø§Ø­ÛŒ ÛŒÚ©Ù¾Ø§Ø±Ú†Ù‡ØŒ Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ Ùˆ Ù‚Ø§Ø¨Ù„ Ù†Ú¯Ù‡Ø¯Ø§Ø±ÛŒ  
**ğŸ“ Ù¾Ø´ØªÛŒØ¨Ø§Ù†ÛŒ:** Ù…Ø±Ø§Ø¬Ø¹Ù‡ Ø¨Ù‡ ÙØ§ÛŒÙ„â€ŒÙ‡Ø§ÛŒ .mdc Ø¨Ø±Ø§ÛŒ Ø±Ø§Ù‡Ù†Ù…Ø§ÛŒ Ú©Ø§Ù…Ù„ 

## ğŸ” **NEW: Advanced Search & Filter System**

The latest addition to our professional standards:

### **ğŸš€ Multi-Word Real-Time Search:**
```javascript
// Auto-applied to ALL search features
const searchSystem = new AdvancedSearchSystem({
    apiUrl: '/api/tags' // Customize per page
});
```

### **âœ… Key Features:**
- **Real-time search** with 200ms debouncing
- **Multi-word support** with space separator
- **AND logic** - all words must match
- **Flexible matching** - partial, case-insensitive
- **Results highlighting** with yellow background
- **Live statistics** showing results count
- **Clear button** for easy reset
- **Enter key navigation** to first result
- **Mobile-responsive** design

### **ğŸ¯ Implementation:**
Every page with search/filter functionality will automatically receive:
- Professional search interface
- Advanced backend filtering
- Highlighted search results
- Mobile-optimized experience
- Keyboard accessibility

---

## ğŸ¯ **Ù‚ÙˆØ§Ù†ÛŒÙ† Ø¬Ø¯ÛŒØ¯ Theme Management**

### **âŒ NEVER DO:**
```javascript
// âŒ Don't create local toggleTheme functions
function toggleTheme() {
    localStorage.setItem('theme', newTheme); // Wrong key!
}

// âŒ Don't use wrong localStorage keys
localStorage.getItem('theme')
localStorage.getItem('samanet_theme') // Typo!
```

### **âœ… ALWAYS DO:**
```javascript
// âœ… Use central theme system
// Theme system handled by theme-system.js - no local override needed

// âœ… Use correct localStorage key
localStorage.getItem('samanat_theme')

// âœ… Use global toggleTheme function
<button onclick="toggleTheme()">Toggle Theme</button>
```

---

## ğŸ“Š **Ø¢Ù…Ø§Ø± Ø§Ø³ØªÙØ§Ø¯Ù‡ Ø§Ø² Ù‚ÙˆØ§Ù†ÛŒÙ†**

### **âœ… Ù…ÙˆÙÙ‚ÛŒØªâ€ŒÙ‡Ø§ÛŒ Ø§Ø®ÛŒØ±:**
- âœ… **ØµÙØ­Ù‡ Ø¯Ø±Ø®ÙˆØ§Ø³Øªâ€ŒÙ‡Ø§:** Ø¨Ø§Ø²Ø·Ø±Ø§Ø­ÛŒ Ú©Ø§Ù…Ù„ Ù…Ø·Ø§Ø¨Ù‚ dashboard.css
- âœ… **Theme System:** ÛŒÚ©Ù¾Ø§Ø±Ú†Ù‡â€ŒØ³Ø§Ø²ÛŒ Ú©Ø§Ù…Ù„
- âœ… **Ù…Ø´Ú©Ù„ Function Redeclaration:** Ø­Ù„ Ø´Ø¯Ù‡
- âœ… **CSS Variables:** Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯ Ø´Ø¯Ù‡

### **ğŸ“ˆ Ù¾ÛŒØ´Ø±ÙØª Ù¾Ø±ÙˆÚ˜Ù‡:**
- **ÙØ§Ø² 1:** âœ… Dashboard Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ
- **ÙØ§Ø² 2:** âœ… Ù…Ø¯ÛŒØ±ÛŒØª Ú©Ø§Ø±Ø¨Ø±Ø§Ù†
- **ÙØ§Ø² 3:** âœ… Ø¯Ø±Ø®ÙˆØ§Ø³Øªâ€ŒÙ‡Ø§ÛŒ Ø­ÙˆØ§Ù„Ù‡
- **ÙØ§Ø² 4:** ğŸš§ Ø¯Ø± Ø­Ø§Ù„ ØªÙˆØ³Ø¹Ù‡

---

## ğŸ”§ **Ù†Ø­ÙˆÙ‡ Ø§Ø³ØªÙØ§Ø¯Ù‡**

### **Ø¨Ø±Ø§ÛŒ ØªÙˆØ³Ø¹Ù‡â€ŒØ¯Ù‡Ù†Ø¯Ú¯Ø§Ù†:**
1. Ù‡Ù…ÛŒØ´Ù‡ Ø§Ø² `MANDATORY-DESIGN.mdc` Ø´Ø±ÙˆØ¹ Ú©Ù†ÛŒØ¯
2. Ù‚ÙˆØ§Ù†ÛŒÙ† `samanat-coding-standard.mdc` Ø±Ø§ Ø±Ø¹Ø§ÛŒØª Ú©Ù†ÛŒØ¯
3. Ø¯Ø± ØµÙˆØ±Øª Ù…Ø´Ú©Ù„ØŒ `DEBUGGING-RULES.mdc` Ø±Ø§ Ù…Ø·Ø§Ù„Ø¹Ù‡ Ú©Ù†ÛŒØ¯
4. Ø¨Ø±Ø§ÛŒ Ø·Ø±Ø§Ø­ÛŒ Ù…Ø¯Ø±Ù†ØŒ `ultra-modern-design-system.mdc` Ø±Ø§ Ø¨Ø¨ÛŒÙ†ÛŒØ¯

### **Ø¨Ø±Ø§ÛŒ Cursor AI:**
- Ù‡Ù…Ù‡ Ù‚ÙˆØ§Ù†ÛŒÙ† Ø¨ØµÙˆØ±Øª Ø®ÙˆØ¯Ú©Ø§Ø± Ø§Ø¹Ù…Ø§Ù„ Ù…ÛŒâ€ŒØ´ÙˆÙ†Ø¯
- Ø¯Ø± ØµÙˆØ±Øª ØªØ¯Ø§Ø®Ù„ØŒ Ø§ÙˆÙ„ÙˆÛŒØª Ø¨Ø§ `MANDATORY-DESIGN.mdc` Ø§Ø³Øª
- Ù‡Ø± ØµÙØ­Ù‡ Ø¬Ø¯ÛŒØ¯ Ø¨Ø§ÛŒØ¯ Ù…Ø·Ø§Ø¨Ù‚ Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯Ù‡Ø§ Ø¨Ø§Ø´Ø¯

---

## ğŸ“ **Ú†Ú©â€ŒÙ„ÛŒØ³Øª ØªÙˆØ³Ø¹Ù‡**

### **Ù‚Ø¨Ù„ Ø§Ø² Ø´Ø±ÙˆØ¹ Ù‡Ø± ØµÙØ­Ù‡:**
- [ ] Ù…Ø·Ø§Ù„Ø¹Ù‡ `MANDATORY-DESIGN.mdc`
- [ ] Ø¨Ø±Ø±Ø³ÛŒ Ù†Ù…ÙˆÙ†Ù‡â€ŒÙ‡Ø§ÛŒ Ù…ÙˆØ¬ÙˆØ¯
- [ ] Ø¢Ù…Ø§Ø¯Ù‡â€ŒØ³Ø§Ø²ÛŒ CSS variables
- [ ] ØªØ¹Ø±ÛŒÙ Ø³Ø§Ø®ØªØ§Ø± HTML

### **Ø­ÛŒÙ† ØªÙˆØ³Ø¹Ù‡:**
- [ ] Ø±Ø¹Ø§ÛŒØª Ú©Ù„Ø§Ø³â€ŒÙ‡Ø§ÛŒ Ø§Ù„Ø²Ø§Ù…ÛŒ
- [ ] ØªØ³Øª responsive design
- [ ] Ø¨Ø±Ø±Ø³ÛŒ dark mode
- [ ] ØªØ³Øª theme persistence

### **Ù¾Ø³ Ø§Ø² ØªÚ©Ù…ÛŒÙ„:**
- [ ] ØªØ³Øª Ø¯Ø± Ù…Ø±ÙˆØ±Ú¯Ø±Ù‡Ø§ÛŒ Ù…Ø®ØªÙ„Ù
- [ ] Ø¨Ø±Ø±Ø³ÛŒ console errors
- [ ] ØªØ³Øª accessibility
- [ ] Ù…Ø³ØªÙ†Ø¯Ø³Ø§Ø²ÛŒ ØªØºÛŒÛŒØ±Ø§Øª

---

## ğŸš€ **Ø¨Ø±Ù†Ø§Ù…Ù‡ Ø¢ÛŒÙ†Ø¯Ù‡**

### **Ù†Ø³Ø®Ù‡ 7.0 (Ø¢ÛŒÙ†Ø¯Ù‡ Ù†Ø²Ø¯ÛŒÚ©):**
- Ù‚ÙˆØ§Ù†ÛŒÙ† Ù…Ø¯ÛŒØ±ÛŒØª Ø§Ø³Ù†Ø§Ø¯
- Ø§Ø³ØªØ§Ù†Ø¯Ø§Ø±Ø¯Ù‡Ø§ÛŒ API
- Ø±Ø§Ù‡Ù†Ù…Ø§ÛŒ Performance

### **Ù†Ø³Ø®Ù‡ 8.0 (Ø¢ÛŒÙ†Ø¯Ù‡ Ù…ÛŒØ§Ù†Ù‡):**
- Ù‚ÙˆØ§Ù†ÛŒÙ† Ø³ÛŒØ³ØªÙ… ØªØ§ÛŒÛŒØ¯
- Workflow Management
- Advanced Components

---

**ğŸ¯ Ù‡Ø¯Ù Ù†Ù‡Ø§ÛŒÛŒ:** Ø³ÛŒØ³ØªÙ… Ø·Ø±Ø§Ø­ÛŒ ÛŒÚ©Ù¾Ø§Ø±Ú†Ù‡ØŒ Ø­Ø±ÙÙ‡â€ŒØ§ÛŒ Ùˆ Ù‚Ø§Ø¨Ù„ Ù†Ú¯Ù‡Ø¯Ø§Ø±ÛŒ  
**ğŸ“ Ù¾Ø´ØªÛŒØ¨Ø§Ù†ÛŒ:** Ù…Ø±Ø§Ø¬Ø¹Ù‡ Ø¨Ù‡ ÙØ§ÛŒÙ„â€ŒÙ‡Ø§ÛŒ .mdc Ø¨Ø±Ø§ÛŒ Ø±Ø§Ù‡Ù†Ù…Ø§ÛŒ Ú©Ø§Ù…Ù„ 